docs/tasks/administer-cluster/safely-drain-node.md

```diff
@@ -0,0 +1,84 @@
+---
+assignees:
+- davidopp
+
+---
+ 
+{% capture overview %}
+This page shows how to safely drain a machine, respecting the application-level
+disruption SLOs you have specified using PodDisruptionBudget.
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* We assume you are using Kubernetes release >= 1.5.
+
+* We assume you have created [PodDisruptionBudget(s)](../../admin/disruptions.md) to express the
+application-level disruption SLOs you want the system to enforce.
+
+{% endcapture %}
+
+{% capture steps %}
+
+### Use `kubectl drain` to remove a node from service
+
+You can use `kubectl drain` to safely evict all of your pods from a node before you perform maintenance
+on the node (e.g. kernel upgrade, hardware maintenance, etc.). To define a few terms:
+
+* **your pods**: Certain system pods that cannot be killed are ignored; see the
+[kubectl drain](../../user-guide/kubectl/kubectl_drain.md) documentation for more details
+* **safely**: (1) the evictions wil give the containers [graceful termination](../../user-guide/production-pods.md#lifecycle-hooks-and-termination-notice)
+and (2) will respect the PodDisruptionBudgets
+
+What this means is that when `kubectl drain` returns (without giving an error), all of
```

Comment:
"When `kubectl drain` returns successfully, that indicates that all of your pods have been safely evicted (respecting the desired graceful termination period, and without violating any application-level disruption SLOs). It is then safe to bring down the node by powering down a physical machine, deleting a cloud platform virtual machine, or equivalent operation."